lib/server/services/shift/ShiftService.js

```diff
@@ -62,8 +65,41 @@ class ShiftService {
             shift,
             issuesLogs: await this.getShiftIssues(shift.start, shiftType),
             typeSpecific: null,
+            infoFromPreviousShifter: { value: '' },
         };
 
+        // Find EoS reports from the last shift with the correct title
+        const pastShift = getShiftFromTimestamp(shift.start - SHIFT_DURATION);
+        const expectedPreviousReportTitle = formatEosReportTitle(pastShift, shiftType);
+
+        const previousShiftReports = await getLogsByTitle(expectedPreviousReportTitle, { rootOnly: true });
+
+        if (previousShiftReports.length === 0) {
+            ret.infoFromPreviousShifter.errorMessage = `No ${shiftType} EoS reports found from the previous shift`;
+        } else if (previousShiftReports.length > 1) {
+            ret.infoFromPreviousShifter.errorMessage = `Multiple ${shiftType} EoS reports found from the previous shift`;
+        } else {
+            const eosReportText = previousShiftReports[0].text;
+            if (!eosReportText) {
+                ret.infoFromPreviousShifter.errorMessage = 'EoS report from the previous shift is missing the text field';
+            } else {
+                const infoFromPreviousShifterRegex = /\n\n### For next shifter\n([\S\s]*?)\n\n###/g;
```

Comment:
I think you can ignore if there is more than one match, we can simply take the first one (it should not happen, and it's not critical at all if we simply ignore the supplementary ones).